### PR TITLE
Add client.Tx.SetTimestamp.

### PR DESCRIPTION
**Pull request by petermattis**
Added the ability to set the timestamp for read/write operations within
a transaction.

Migrate some more usages of client.KV to client.DB.

See #997.

### CODE REVIEW

File: client/db.go [hunk 1]
```diff
@@ -74,10 +74,8 @@ func (kv *KeyValue) setValue(v *proto.Value) {
 	} else if v.Integer != nil {
 		kv.Value = v.Integer
 	}
-	if ts := v.Timestamp; ts != nil {
-		sec := ts.WallTime / 1e9
-		nsec := ts.WallTime % 1e9
-		kv.Timestamp = time.Unix(sec, nsec)
+	if v.Timestamp != nil {
```
Comment (tamird) [inline client/db.go]: y no extract local?

File: kv/txn_test.go [hunk 2]
```diff
@@ -215,28 +181,22 @@ func verifyUncertainty(concurrency int, maxOffset time.Duration, t *testing.T) {
...
-				if err := txn.Run(client.Call{
-					Args: &proto.GetRequest{
-						RequestHeader: proto.RequestHeader{
-							Key:       key,
-							Timestamp: futureTS,
-						},
-					},
-					Reply: &gr,
-				}); err != nil {
+				tx.SetTimestamp(futureTS.GoTime())
```
Comment (spencerkimball) [inline kv/txn_test.go]: I think we should remove SetTimestamp here. It's not something we should support.

File: client/txn.go [hunk 3]
```diff
@@ -46,6 +46,7 @@ type txnSender struct {
 func (ts *txnSender) Send(ctx context.Context, call Call) {
 	// Send call through wrapped sender.
 	call.Args.Header().Txn = &ts.txn
+	call.Args.Header().Timestamp = ts.opTimestamp
```
Comment (spencerkimball) [inline client/txn.go]: This doesn't really make sense. You shouldn't be able to change the timestamp within a transaction. The transaction timestamp should take precedence.

File: client/db.go (hunk 1 above)
Comment (petermattis) [inline client/db.go]: You mean the `ts := v.Timestamp`? That didn't seem as useful once most of the work was moved to `GoTime()`.

File: client/txn.go (hunk 3 above)
Comment (petermattis) [inline client/txn.go]: I'm just copying what the previous code was doing.

File: kv/txn_test.go (hunk 2 above)
Comment (petermattis) [inline kv/txn_test.go]: So how would you suggest modifying this test so that it behaves similarly to before?

Comment (petermattis) [inline kv/txn_test.go]: Lovely. Commenting this out and the test still passes. I'll remove the `SetTimestamp` stuff.

Comment (spencerkimball) [inline kv/txn_test.go]: This part of the test is just verifying that within the same txn, you never run into uncertainty interval errors. I don't see this is a particularly useful or realistic part of the test. @tschottdorf ?

**Comment by petermattis** [comment]
This needs another look. Specifically the `client/Tx.InternalSetPriority` stuff which is needed by some of our existing tests.

**Comment by spencerkimball** [comment]
LGTM

**Pull request merged**